examples/managers/mid_agreement_payments.py

```diff
@@ -0,0 +1,110 @@
+import asyncio
+import logging.config
+from datetime import datetime, timedelta
+
+from golem.managers import (
+    Buffer,
+    DefaultAgreementManager,
+    DefaultProposalManager,
+    MidAgreementPaymentsNegotiator,
+    NegotiatingPlugin,
+    PayAllPaymentManager,
+    PaymentPlatformNegotiator,
+    RefreshingDemandManager,
+    SequentialWorkManager,
+    SingleUseActivityManager,
+    WorkContext,
+    WorkResult,
+)
+from golem.node import GolemNode
+from golem.payload import RepositoryVmPayload
+from golem.utils.logging import DEFAULT_LOGGING
+
+WORK_RUN_TIME = timedelta(minutes=60)
+
+
+async def print_yagna_payment_status(yagna_path="yagna", network="goerli"):
+    process = await asyncio.create_subprocess_exec(
+        yagna_path,
+        "payment",
+        "status",
+        "--network",
+        network,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+
+    stdout, stderr = await process.communicate()
+
+    if process.returncode != 0:
+        raise Exception(
+            f"Exited with code `{process.returncode!r}`!\nstdout:\n{stdout!r}\nstderr:\n{stderr!r}"
+        )
+
+    print(stdout.decode(), flush=True)
+
+
+async def idle_work(context: WorkContext) -> str:
+    work_start_time = datetime.utcnow()
+    result = ""
+    while datetime.utcnow() - work_start_time <= WORK_RUN_TIME:
+        r = await context.run("echo 'hello golem'")
+        await r.wait()
+        for event in r.events:
+            result += event.stdout
+
+        await asyncio.sleep(60)
+        await print_yagna_payment_status()
+
+    return result
+
+
+async def main():
+    logging.config.dictConfig(DEFAULT_LOGGING)
+    payload = RepositoryVmPayload("9a3b5d67b0b27746283cb5f287c13eab1beaa12d92a9f536b747c7ae")
+    await print_yagna_payment_status()
+
+    golem = GolemNode()
+
+    payment_manager = PayAllPaymentManager(golem, budget=10.0)
+    demand_manager = RefreshingDemandManager(
+        golem,
+        payment_manager.get_allocation,
+        [
+            payload,
+        ],
+    )
+    proposal_manager = DefaultProposalManager(
+        golem,
+        demand_manager.get_initial_proposal,
+        plugins=[
+            NegotiatingPlugin(
+                proposal_negotiators=[
+                    PaymentPlatformNegotiator(),
+                    MidAgreementPaymentsNegotiator(
+                        requested_debit_note_interval=timedelta(minutes=3),
+                        requested_payment_timeout=timedelta(minutes=30),
```

Comment:
shouldn't this be longer? like e.g. 12 or 24 hours? 30 minutes sound very short ...

Reply:
This is lower for the example